dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerVectorStoreRecordCollection.cs

```
@@ -0,0 +1,342 @@
+
+    private static Dictionary<string, object?> Map(TRecord record, VectorStoreRecordPropertyReader propertyReader)
+    {
+        Dictionary<string, object?> map = new(StringComparer.Ordinal);
+        map[propertyReader.KeyProperty.DataModelPropertyName] = propertyReader.KeyPropertyInfo.GetValue(record);
+
+        var dataProperties = propertyReader.DataProperties;
+        for (int i = 0; i < dataProperties.Count; i++)
+        {
+            object value = propertyReader.DataPropertiesInfo[i].GetValue(record);
+            // SQL Server does not support arrays, so we need to serialize them to JSON.
+            object? mappedValue = value switch
+            {
+                string[] array => JsonSerializer.Serialize(array),
+                List<string> list => JsonSerializer.Serialize(list),
+                _ => value
+            };
+
+            map[dataProperties[i].DataModelPropertyName] = mappedValue;
+        }
+        var vectorProperties = propertyReader.VectorProperties;
+        for (int i = 0; i < vectorProperties.Count; i++)
+        {
+            // We restrict the vector properties to ReadOnlyMemory<float> so the cast here is safe.
+            ReadOnlyMemory<float> floats = (ReadOnlyMemory<float>)propertyReader.VectorPropertiesInfo[i].GetValue(record);
+            // We know that SqlServer supports JSON serialization, so we can serialize the vector as JSON now,
+            // so the SqlServerCommandBuilder does not need to worry about that.
+            // TODO adsitnik perf: we could remove the dependency to System.Text.Json
+            // by using a hand-written serializer.
+            map[vectorProperties[i].DataModelPropertyName] = JsonSerializer.Serialize(floats);
+        }
+
+        return map;
+    }
+
+    private static TRecord Map(SqlDataReader reader, VectorStoreRecordPropertyReader propertyReader)
+    {
+        TRecord record = Activator.CreateInstance<TRecord>()!;
```

The POCO mapper should verify on construction that TRecord has a parameterless constructor so that it fails early and not only at mapping time.  See e.g. https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.Memory.Qdrant/QdrantVectorStoreRecordMapper.cs#L34